doc/developers/contributing.rst

```diff
@@ -790,17 +790,66 @@ In the following example, k is deprecated and renamed to n_clusters::
 
     import warnings
 
-    def example_function(n_clusters=8, k=None):
-        if k is not None:
+    def example_function(n_clusters=8, k='deprecated'):
+        if k != 'deprecated':
             warnings.warn("'k' was renamed to n_clusters in version 0.13 and "
                           "will be removed in 0.15.", DeprecationWarning)
             n_clusters = k
 
+When the change is in a class, we conduct validate and raise warning in ``fit``::
+
+  import warnings
+
+  class ExampleEstimator:
+      def __init__(self, n_clusters=8, k='deprecated'):
+          self.n_clusters = n_clusters
+          self.k = k
+
+      def fit(self, X, y):
+          if k != 'deprecated':
+              warnings.warn("'k' was renamed to n_clusters in version 0.13 and "
+                            "will be removed in 0.15.", DeprecationWarning)
+              self._n_clusters = k
+          else:
+              self._n_clusters = self.n_clusters
+
+
+If the default value of a parameter needs to be changed, please replace the
+default value with a specific value (e.g., ``warn``) and raise
+``FutureWarning`` when users are using the default value. In the following
+example, we change the default value of ``n_clusters`` from 5 to 10
+(current version is 0.20). You can also refer to merged PRs
+(e.g., `#10331 <https://github.com/scikit-learn/scikit-learn/pull/10331>`__
```

Comment:
do these add anything? I mean I'm not super opposed but I'm not sure what's in the PR that's not here.

Reply:
I would remove the PR references. I imagine the discussion is a bit chaotic compared to the guidelines we are trying to provide.